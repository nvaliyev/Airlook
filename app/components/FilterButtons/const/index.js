 const ButtonsList = ['Yesterday', 'Today', 'All Flights', 'Tomorrow']

export default ButtonsList;