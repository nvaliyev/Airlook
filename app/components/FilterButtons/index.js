
import React, { useState } from 'react';
import styles from './style';
import { ButtonGroup } from 'react-native-elements';
import ButtonsList from './const';

const FilterButtons = () => {

  const [selectedIndex, updateIndex] = useState([0]);

  return (
    <ButtonGroup
      onPress={updateIndex}
      selectedIndexes={selectedIndex}
      textStyle={styles.text}
      selectedButtonStyle={styles.selectedText}
      buttonStyle={styles.button}
      buttons={ButtonsList}
      containerStyle={styles.container}
      selectMultiple={true}
      innerBorderStyle={styles.innerBorder}
    />
  )
}
export default FilterButtons;