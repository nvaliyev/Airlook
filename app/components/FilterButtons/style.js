import {StyleSheet} from 'react-native';
import config  from '../../servises/utils';

 const styles = StyleSheet.create({
     container:{
      flex:1,
      alignItems:'center',
      maxHeight:30,
      flexDirection:'row',
      alignSelf:'center',
      justifyContent:'center',
      backgroundColor:'transparent',
      opacity:0.9,
      borderWidth:0,
      padding:1,
     },
     selectedText:{
    backgroundColor:config.eminence,
    elevation:3,
     },
     innerBorder:{
        width:0,
        color:config.lilac,
        backgroundColor:config.lilac
        
     },
    button: {
        backgroundColor:config.lilac,
        borderRadius:4,
        borderWidth:1,
        borderColor:config.eminence,
        marginHorizontal:2,        
    },
    text: 
        {
         color:config.eminence,
            fontSize:12
        }

    
});
export default styles
