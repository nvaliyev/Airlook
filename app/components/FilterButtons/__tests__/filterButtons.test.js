import 'react-native';
import React from './node_modules/react';
import ShallowRenderer from './node_modules/react-test-renderer/shallow';
import FilterButtons from '..';
describe(' FilterButtons ', () => {
    test(' FilterButtons snapshot', () => {
        const renderer = new ShallowRenderer();
        const snapshot = renderer.render(< FilterButtons/>);
        expect(snapshot).toMatchSnapshot();
    });
});


