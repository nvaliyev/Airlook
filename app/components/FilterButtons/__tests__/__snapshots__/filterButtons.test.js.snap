// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` FilterButtons   FilterButtons snapshot 1`] = `
<Themed.ButtonGroup
  buttonStyle={
    Object {
      "backgroundColor": "#F7F9FC",
      "borderColor": "#6f347c",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginHorizontal": 2,
    }
  }
  buttons={
    Array [
      "Yesterday",
      "Today",
      "All Flights",
      "Tomorrow",
    ]
  }
  containerStyle={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "transparent",
      "borderWidth": 0,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "maxHeight": 30,
      "opacity": 0.9,
      "padding": 1,
    }
  }
  innerBorderStyle={
    Object {
      "backgroundColor": "#F7F9FC",
      "color": "#F7F9FC",
      "width": 0,
    }
  }
  onPress={[Function]}
  selectMultiple={true}
  selectedButtonStyle={
    Object {
      "backgroundColor": "#6f347c",
      "elevation": 3,
    }
  }
  selectedIndexes={
    Array [
      0,
    ]
  }
  textStyle={
    Object {
      "color": "#6f347c",
      "fontSize": 12,
    }
  }
/>
`;
